### ISSUE TITLE
re and im should work for matrix expressions

### ISSUE THREAD

**Issue by asmeurer** [sympy/sympy#11600]
Right now

```
In [3]: re(MatrixSymbol('x', n, m))
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-3-fb9d29ea43cf> in <module>()
----> 1 re(MatrixSymbol('x', n, m))

/Users/aaronmeurer/Documents/Python/sympy/sympy/sympy/core/function.py in __new__(cls, *args, **options)
    425
    426         evaluate = options.get('evaluate', global_evaluate[0])
--> 427         result = super(Function, cls).__new__(cls, *args, **options)
    428         if not evaluate or not isinstance(result, cls):
    429             return result

/Users/aaronmeurer/Documents/Python/sympy/sympy/sympy/core/function.py in __new__(cls, *args, **options)
    249
    250         if evaluate:
--> 251             evaluated = cls.eval(*args)
    252             if evaluated is not None:
    253                 return evaluated

/Users/aaronmeurer/Documents/Python/sympy/sympy/sympy/functions/elementary/complexes.py in eval(cls, arg)
     66             args = Add.make_args(arg)
     67             for term in args:
---> 68                 coeff = term.as_coefficient(S.ImaginaryUnit)
     69
     70                 if coeff is not None:

AttributeError: 'MatrixSymbol' object has no attribute 'as_coefficient'
```

We may need to create special matrix `re` and `im` classes. Or, more generally, is there a matrix expression class to represent calling a scalar function on all the elements?

**Comment by ArifAhmed1995** [comment]
I've briefly looked through the `matrices` module and can't find any such classes.I'll work on adding such classes to `matexpr`

**Comment by nikoskaragiannakis** [comment]
Could this be related to https://github.com/sympy/sympy/issues/11599 ?

**Comment by asmeurer** [comment]
This works for me in master. Should this be closed?

**Comment by ArifAhmed1995** [comment]
Yeah I think so. [#11696](https://github.com/sympy/sympy/pull/11696) fixes it.

**Comment by asmeurer** [comment]
OK. In the future write "fixes #11600" in that exact format in the pull request description so that GitHub automatically closes the issue when the PR is merged.
